soundbay/utils/conf_validator.py

```diff
@@ -0,0 +1,132 @@
+from typing import Optional
+from pydantic import BaseModel, validator
+
+
+class Dataset(BaseModel):
```

Comment:
this is up for debate (a design choice): perhaps we should validate more arguments here, rather than down the line? (for example - min_freq_filtering has its own validation within the method that uses this argument, far ahead in the running of the model)

obviously we wouldnt want to validate every single config argument, im just wondering if more validations should be added here, instead of typecheck and value checks deeper inside the code.

*I think I'd define the validation for sample rate and data_sample_rate here as well, for example

Reply:
Added your suggestsions